workos/directory_sync.py

```diff
@@ -19,86 +19,76 @@ def request_helper(self):
             limit (int): Maximum number of records to return.
-            before (str): Pagination cursor to receive records before a provided Directory Endpoint ID.
-            after (str): Pagination cursor to receive records after a provided Directory Endpoint ID.
+            before (str): Pagination cursor to receive records before a provided Directory ID.
+            after (str): Pagination cursor to receive records after a provided Directory ID.
 
         Returns:
             dict: Directory Users response from WorkOS.
         """
         params = {"limit": limit, "before": before, "after": after}
+        if group is not None:
+            params["group"] = group
+        if directory is not None:
+            params["directory"] = directory
         return self.request_helper.request(
-            "directories/{directory_endpoint_id}/users".format(
-                directory_endpoint_id=directory_endpoint_id
-            ),
+            "directory_users",
             method=REQUEST_METHOD_GET,
             params=params,
             token=workos.api_key,
         )
 
     def list_groups(
-        self, directory_endpoint_id, limit=RESPONSE_LIMIT, before=None, after=None
+        self, directory=None, user=None, limit=RESPONSE_LIMIT, before=None, after=None
     ):
-        """Gets a list of provisioned Groups for a Directory Endpoint.
+        """Gets a list of provisioned Groups for a Directory .
+
+        Note, either 'directory' or 'user' must be provided.
 
         Args:
-            directory_endpoint_id (str): Directory Endpoint unique identifier.
+            directory (str): Directory unique identifier.
+            user (str): Directory User unique identifier.
             limit (int): Maximum number of records to return.
-            before (str): Pagination cursor to receive records before a provided Directory Endpoint ID.
-            after (str): Pagination cursor to receive records after a provided Directory Endpoint ID.
+            before (str): Pagination cursor to receive records before a provided Directory ID.
+            after (str): Pagination cursor to receive records after a provided Directory ID.
 
         Returns:
             dict: Directory Groups response from WorkOS.
         """
         params = {"limit": limit, "before": before, "after": after}
+        if user is not None:
+            params["user"] = user
+        if directory is not None:
+            params["directory"] = directory
         return self.request_helper.request(
-            "directories/{directory_endpoint_id}/groups".format(
-                directory_endpoint_id=directory_endpoint_id
-            ),
+            "directory_groups",
             method=REQUEST_METHOD_GET,
             params=params,
             token=workos.api_key,
         )
 
-    def get_user(self, directory_endpoint_id, directory_user_id):
+    def get_user(self, directory, directory_user):
```

Comment:
I have no in depth knowledge of the app side changes or dsync in general but... is getting a user not doable via the `directory_users` endpoint?

Reply:
You definitely can, but the `directory_users` endpoint gives back a paginated list. I think it's still good to have a way to retrieve a single record.